Engine/main/update.cpp

```diff
@@ -81,48 +82,48 @@ int do_movelist_move(short*mlnum,int*xx,int*yy) {
 
     int adjAmnt = 3;
     // 2.70: if the X permove is also <=1, don't do the skipping
-    if (((xpermove & 0xffff0000) == 0xffff0000) ||
-        ((xpermove & 0xffff0000) == 0x00000000))
+    if ((trunc(xpermove) == -1) ||
+        (trunc(xpermove) == 0) )
       adjAmnt = 2;
 
     // 2.61 RC1: correct this to work with > -1 as well as < 1
-    if (ypermove == 0) { }
+    if (fabs(ypermove) < delta) { }
     // Y per move is < 1, so finish the move
-    else if ((ypermove & 0xffff0000) == 0)
+    else if (trunc(ypermove) == 0)
       targety -= adjAmnt;
     // Y per move is -1 exactly, don't snap to finish
-    else if (ypermove == 0xffff0000) { }
+    else if (fabs(ypermove - 1.0) < delta) { }
     // Y per move is > -1, so finish the move
-    else if ((ypermove & 0xffff0000) == 0xffff0000)
+    else if (trunc(ypermove) == -1)
       targety += adjAmnt;
   }
-  else xps=cmls->fromx+(int)(fixtof(xpermove)*(float)cmls->onpart);
+  else xps=cmls->fromx+(int)(xpermove*(float)cmls->onpart);
 
   if (cmls->doneflag & 2) {
     // Y-movement has finished
 
     int adjAmnt = 3;
 
     // if the Y permove is also <=1, don't skip as far
-    if (((ypermove & 0xffff0000) == 0xffff0000) ||
-        ((ypermove & 0xffff0000) == 0x00000000))
+    if ((trunc(ypermove) == -1) ||
+        (trunc(ypermove) == 0))
       adjAmnt = 2;
 
-    if (xpermove == 0) { }
+    if (fabs(xpermove) < delta) { }
     // Y per move is < 1, so finish the move
-    else if ((xpermove & 0xffff0000) == 0)
+    else if (trunc(xpermove) == 0)
       targetx -= adjAmnt;
     // X per move is -1 exactly, don't snap to finish
-    else if (xpermove == 0xffff0000) { }
+    else if (fabs(xpermove -1) < delta) { }
```

Comment:
Same as above, should have double negation:
```fabs(xpermove - (-1.0)) < delta```

Reply:
I'm tempted to use the almost_equal function, but I don't know what value of ULPs we'd need for equality, or even what "equality" means in this specific context.

Reply:
Hmm maybe make a local function here for now, with the chosen delta constant. The point is rather to reduce possible mistakes when comparing, since you will be using a function instead.

Reply:
Redone with local function